File: lib/models/tv_categories.dart
import 'package:video_player/models/tv_channel.dart';

class TvCategories {
  const TvCategories({required this.id, required this.title, required this.tvChannels});

  final String id;
  final String title;
  final List<TvChannel> tvChannels;

  Map<String, dynamic> toMap() => {'id': id, 'title': title, 'tvChannels': tvChannels.map((v) => v.toJson()).toList()};

  @override
  String toString() => 'TvCategories{id: $id, title: $title}';
}

File: android/src/main/kotlin/uz/shs/video_player/models/TvCategories.kt
package uz.shs.video_player.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class TvCategories(
    @SerializedName("id")
    val id: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("tvChannels")
    val channels: List<TvChannel>,
) : Serializable